File: LR2/27/27.java
import java.util.Scanner;

public class Main {

    public static boolean correctDate(int day, int month) {
        return day >= 1 && day <= 31 && month >= 1 && month <= 12;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int nums = scanner.nextInt();
        
        for (int i = 0; i < nums; i++) {
            String token = scanner.next();
            boolean validInput = true;
            
            // Проверка, состоит ли входная строка только из цифр
            for (int j = 0; j < token.length(); j++) {
                if (!Character.isDigit(token.charAt(j))) {
                    validInput = false;
                    break;
                }
            }
            
            if (!validInput || token.length() < 2) {
                System.out.print("NO");
                continue;
            }
            
            boolean isCorrectDate = false;
            int number = Integer.parseInt(token);
            
            if (number >= 11 && number < 100) {
                int day = number / 10;
                int month = number % 10;
                isCorrectDate = correctDate(day, month);
            } else if (number >= 100 && number < 1000) {
                int day = number / 10;
                int month = number % 10;
                isCorrectDate = correctDate(day, month);
                if (!isCorrectDate) {
                    day = number / 100;
                    month = number % 100;
                    isCorrectDate = correctDate(day, month);
                }
            } else if (number >= 1000 && number <= 3112) {
                int day = number / 100;
                int month = number % 100;
                isCorrectDate = correctDate(day, month);
            }
            
            System.out.print(isCorrectDate ? "YES" : "NO");
        }
        
        scanner.close();
    }
}

File: LR2/27/27.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"unicode"
)

func correctDate(day, month int) bool {
	return day >= 1 && day <= 31 && month >= 1 && month <= 12
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	// Считываем количество чисел
	scanner.Scan()
	nums, _ := strconv.Atoi(scanner.Text())

	for i := 0; i < nums; i++ {
		scanner.Scan()
		token := scanner.Text()

		valid := true
		// Проверяем, все ли символы в токене являются цифрами
		for _, c := range token {
			if !unicode.IsDigit(c) {
				valid = false
				break
			}
		}
		if !valid || len(token) < 2 {
			fmt.Print("NO")
			continue
		}

		number, _ := strconv.Atoi(token)
		isCorrect := false

		if number >= 11 && number < 100 {
			day := number / 10
			month := number % 10
			isCorrect = correctDate(day, month)
		} else if number >= 100 && number < 1000 {
			day := number / 10
			month := number % 10
			isCorrect = correctDate(day, month)
			if !isCorrect {
				day = number / 100
				month = number % 100
				isCorrect = correctDate(day, month)
			}
		} else if number >= 1000 && number <= 3112 {
			day := number / 100
			month := number % 100
			isCorrect = correctDate(day, month)
		}

		if isCorrect {
			fmt.Print("YES")
		} else {
			fmt.Print("NO")
		}
	}
}
